include/kframework/evm.md

```diff
@@ -2286,8 +2289,9 @@ There are several helpers for calculating gas (most of them also specified in th
  // --------------------------------------------------------------------------------------------------
     rule #accountEmpty(CODE, NONCE, BAL) => CODE ==K .Bytes andBool NONCE ==Int 0 andBool BAL ==Int 0
 
-    syntax Int ::= #allBut64th ( Int ) [function, total, smtlib(gas_allBut64th)]
- // ----------------------------------------------------------------------------
+    syntax Gas ::= #allBut64th ( Gas ) [function, total, smtlib(gas_allBut64th_Gas)]
+    syntax Int ::= #allBut64th ( Int ) [function, total, smtlib(gas_allBut64th_Int)]
+ // --------------------------------------------------------------------------------
     rule [allBut64th.pos]: #allBut64th(N) => N -Int (N /Int 64) requires 0 <=Int N
     rule [allBut64th.neg]: #allBut64th(N) => 0                  requires N  <Int 0
```

Comment:
Do we actually need 2 productions for this? Can we just have the `Gas -> Gas` variant? The rules below will still work directly for that, because `Int` is a subsort, and `N` in the rules is restricted to being an `Int` by the requires clauses. You can add the extra sort constraint just to be sure, but I think it should work without both productions.

```suggestion
    syntax Gas ::= #allBut64th ( Gas ) [function, total, smtlib(gas_allBut64th)]
 // ----------------------------------------------------------------------------
    rule [allBut64th.pos]: #allBut64th(N:Int) => N -Int (N /Int 64) requires 0 <=Int N
    rule [allBut64th.neg]: #allBut64th(N:Int) => 0                  requires N  <Int 0
```

Reply:
I run into Sort errors when I strip this back. 
`rule #allBut64th(#gas(G)) => #gas(#allBut64th(G:Int))` errors on RHS as `#allBut64th` returns a `Gas`, in reality this is an `Int` for this case. Is there a way to communicate that aside from duplicating the rule?